pkg/progress/progress_watcher.go

```diff
@@ -0,0 +1,124 @@
+package progress
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Watcher struct {
+	progressPercents int
+	exitCode         int
+	buf              *bufio.Reader
+	subscribers      []func(ProgressState)
+	done             chan bool
+	cmd              *exec.Cmd
+}
+
+func CmdWithProgressWatching(name string, arg ...string) (*Watcher, error) {
+	cmd := exec.Command(name, arg...)
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Watcher{
+		progressPercents: 0,
+		buf:              bufio.NewReader(stdout),
+		subscribers:      []func(ProgressState){},
+		done:             make(chan bool),
+		cmd:              cmd,
+	}, nil
+}
+
+func (p *Watcher) Watch() {
+	// track also if exec is not finished to stop automatically
+	go func() {
+		err := p.cmd.Wait()
+
+		exitCode := p.cmd.ProcessState.ExitCode()
+		p.stop(exitCode)
+
+		if err != nil {
+			//TODO Make sure that it's right way to finish the process
```

Comment:
Please address that TODO

Reply:
added `log.Fatal(err) //fatal will call os.Exit(1)`